src/main/java/org/openrewrite/java/logging/ConvertConfiguration.java

```diff
@@ -0,0 +1,90 @@
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.*;
+import org.openrewrite.quark.Quark;
+import org.openrewrite.remote.Remote;
+import org.openrewrite.text.PlainTextParser;
+import org.openrewrite.tree.ParseError;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Converts a logging configuration file from one format to another.
+ */
+@AllArgsConstructor
+public class ConvertConfiguration extends Recipe {
+
+    @Option(displayName = "Pattern for the files to convert",
+            description = "If set, only the files that match this pattern will be converted.",
+            example = "**/log4j.properties",
+            required = false)
+    @Nullable
+    String filePattern;
+
+    @Option(displayName = "Input format",
+            description = "The id of the input logging configuration format. See [Log4j documentation](https://logging.staged.apache.org/log4j/transform/log4j-converter-config.html#formats) for a list of supported formats.",
+            example = "v1:properties")
+    String inputFormat;
+
+    @Option(displayName = "Output format",
+            description = "The id of the output logging configuration format. See [Log4j documentation](https://logging.staged.apache.org/log4j/transform/log4j-converter-config.html#formats) for a list of supported formats.",
+            example = "v2:xml")
+    String outputFormat;
+
+    @Override
+    public String getDisplayName() {
+        return "Convert logging configuration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Converts the configuration of a logging backend from one format to another. For example it can convert a Log4j 1 properties configuration file into a Log4j Core 2 XML configuration file.";
+    }
+
+    private static final ConfigurationConverter converter = ConfigurationConverter.getInstance();
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new FindSourceFiles(filePattern), new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                if (tree instanceof ParseError || tree instanceof Quark || tree instanceof Remote) {
+                    return tree;
+                }
+                if (tree instanceof SourceFile) {
+                    SourceFile sourceFile = (SourceFile) tree;
+                    ByteArrayInputStream inputStream = new ByteArrayInputStream(sourceFile.printAllAsBytes());
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+                    converter.convert(inputStream, inputFormat, outputStream, outputFormat);
+                    String utf8 = new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
+
+                    return PlainTextParser.convert(sourceFile)
```

Comment:
It looks like the output files are actually XML; should we use the XML parser here to read the converted contents such that subsequent recipes can optionally target their contents as part of the same recipe run?

Reply:
This is exactly the main question I had for you. The `log4j-converter-config` artifact can convert to multiple formats (right now `log4j2.xml`, `log4j2.json`, `log4j2.yaml` and `log4j2.properties` are supported).

In the "Log4j 1 to Log4j Core 2" recipe I only use the XML output, but the `ConvertConfiguration` recipe could be used by users for other conversions. Should I make a `switch` here on `outputFormat` and use `PlainText` only if the format is not one of the recognized ones?